cpp/daal/include/algorithms/pca/pca_types.h

```diff
@@ -200,6 +200,7 @@ class DAAL_EXPORT Input : public InputIface
 public:
     Input();
     Input(const Input & other);
+    Input & operator=(const Input & other) = delete;
```

Comment:
Those should not be deleted, as `Input` objects are copyable by design:
```suggestion
    Input & operator=(const Input & other);
```

Please add the implementation (can be `= default;`) in this .cpp file:
https://github.com/uxlfoundation/oneDAL/blob/main/cpp/daal/src/algorithms/pca/pca_input.cpp

The same applies to `DistributedInput` and `ResultImpl`.

Reply:
This should be resolved now.

Reply:
Hi @Vika-F I've resolved the issue about not deleting the assignment operator. Can you take another quick look at the PR when you get a chance?